src/Domain/Contracts/DetailsCarrier.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Domain\Contracts;
+
+use NunoMaduro\PhpInsights\Domain\Details;
+
+/**
+ * @internal
+ */
+interface DetailsCarrier extends HasDetails
```

Comment:
Not sure about naming here, if someone has a better idea :upside_down_face:

Reply:
Could you explain the purpose if the interface? 😄